src/sentry/api/endpoints/project_transaction_threshold.py

```diff
@@ -0,0 +1,102 @@
+import math
+
+from django.db import transaction
+from rest_framework import serializers, status
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.bases.project import ProjectEndpoint, ProjectSettingPermission
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models import project_transaction_threshold
+from sentry.models.transaction_threshold import TRANSACTION_METRICS, ProjectTransactionThreshold
+
+
+class ProjectTransactionThresholdSerializer(serializers.Serializer):
+    threshold = serializers.IntegerField()
+    metric = serializers.CharField()
+
+    def validate_metric(self, metric):
+        for key, value in TRANSACTION_METRICS.items():
+            if value == metric:
+                return key
+
+        raise serializers.ValidationError(f"Invalid transaction metric - {metric}")
+
+    def validate_threshold(self, threshold):
+        return int(math.ceil(threshold / 100)) * 100
+
+
+class ProjectTransactionThresholdEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectSettingPermission,)
+
+    def has_feature(self, project, request):
+        return features.has(
+            "organizations:project-transaction-threshold", project.organization, actor=request.user
+        )
+
+    def get(self, request, project):
+        if not self.has_feature(project, request):
+            return self.respond(status=404)
+
+        try:
+            project_threshold = ProjectTransactionThreshold.objects.get(
+                project=project,
+                organization=project.organization,
+            )
+        except ProjectTransactionThreshold.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        return Response(
+            serialize(
+                project_threshold,
+                request.user,
+                serializer=project_transaction_threshold.ProjectTransactionThresholdSerializer(),
```

Comment:
I don't think you need to specify the serializer here since you registered it below.